qadence_protocols/mitigations/twirl.py

```diff
@@ -67,23 +60,53 @@ def mitigate(
     samples_twirl_den_list = []
     for twirl in twirls:
         layer = [X(i) for i in twirl]
-        block_twirl = chain(circuit.block, kron(*layer))
+        block_twirl = chain(block, kron(*layer))
 
         circ_twirl_num = QuantumCircuit(block_twirl.n_qubits, block_twirl)
-        model_twirl_num = QuantumModel(circuit=circ_twirl_num, backend=backend)
-        samples_twirl_num = model_twirl_num.sample(noise=noise, n_shots=n_shots)[0]
-        samples_twirl_num_list.append(twirl_swap(n_qubits, twirl, samples_twirl_num))
+        model_twirl_num = QuantumModel(circuit=circ_twirl_num, backend=model.backend.backend.name)
+        samples_twirl_num = model_twirl_num.sample(noise=model._noise)[0]
+        samples_twirl_num_list.append(twirl_swap(block_twirl.n_qubits, twirl, samples_twirl_num))
 
         circ_twirl_den = QuantumCircuit(block_twirl.n_qubits, kron(*layer))
-        model_twirl_den = QuantumModel(circuit=circ_twirl_den, backend=backend)
-        samples_twirl_den = model_twirl_den.sample(noise=noise, n_shots=n_shots)[0]
-        samples_twirl_den_list.append(twirl_swap(n_qubits, twirl, samples_twirl_den))
-
-    expectation: float = 0
-
-    for o in observable:
-        expectation_num = compute_exp(n_qubits, samples_twirl_num_list, o)
-        expectation_den = compute_exp(n_qubits, samples_twirl_den_list, o)
-        expectation += expectation_num / expectation_den
+        model_twirl_den = QuantumModel(circuit=circ_twirl_den, backend=model.backend.backend.name)
+        samples_twirl_den = model_twirl_den.sample(noise=model._noise)[0]
+
+        samples_twirl_den_list.append(twirl_swap(block_twirl.n_qubits, twirl, samples_twirl_den))
+
+    # I am creating a dummy circuit that has nothing to do with the protocol
+    # This needs to be changed
+    dummy = QuantumCircuit(
```

Comment:
What is this for exactly ?

Reply:
Ah, right. It feels like this is an overkill. If you only need to call expectations, I would recommend to use the convenience `expectation` function [here](https://github.com/pasqal-io/qadence/blob/acc647205e2f4b028ba7e0f699e03d384dbb1f38/qadence/execution.py#L181) in the `execution` module.  It relieves you from all the QuantumModel boilerplate.

Reply:
This method still expect a circuit from what i see, so I would need to keep circuit still ?

Reply:
Forgot to mark it resolved, Fixed it in the new push